### ISSUE TITLE
New and updated games minor problems

### ISSUE THREAD

**Issue by ZappaUtopia** [libretro/mame2003-plus-libretro#38]
Hi, i've been going through the list of new/updated games, I have checked most of them but this is just a list of minor things I noticed. Using a Raspberry Pi 3.

`brapboys` (buzzing noise)
`backfire` (some corrupted graphics, don't know how to control)
`crswd2bl` (minor sprite flicker) BEST AVAILABLE
`drifto94` (bad sound)
`dynagear` (small black bars at left of screen, nothing major) BEST AVAILABLE
`packbang` (has music but no sound effects, not sure if it's supposed to be like this ?)
`pnyaa` (bad sound)
`shogwarr` (same buzzing noise as brapboys but goes away)
`titlef` (picture is at top left of screen)

Sound samples need to be in the `BIOS/mame2003-plus` directory, they don't work in the `samples` directory.

**Comment by arcadez** [comment]
Thanks for the headsup!!

Let's start with the neogeo games which were always going to be hit and miss as i backported the 
newer code into the ole driver without updating the video and in some cases the sound as to do so would have meant a newer driver and with it lots of roms would have changed.

crswd2bl (minor sprite flicker) 
it does this in every emulator i've tried it in, likely due to it being a hack of a NeoCD game.

mslug5 (doesn't work, crashes ?)
this one might need some protection i left out as i thought only a bootleg used it i'll add in
mslug5prot_r to the neogeo_machine.c and hopefully this will sort it.

pnyaa (bad sound)
i'll take a look at this one but it might just be it needed the sound improvements from MAME80
which we dont have in this core hence the bad sound

svc (corrupted graphics)
I had to change the original rom as i couldn't find it, this meant that it was highly likely
this rom version would not be compatable with the older style neogeo decyption hence
the graphics are not decrypting correctly, just use the bootleg as i hear it's working fine.

samsho5 (red warning screen)
samsh5sp (red warning screen)
I think i know the problem im guessing it's something about illegal rom carts..?? i'll sort this
now mame2003-plus is missing a couple of defines in neogeo_machine

**Comment by ZappaUtopia** [comment]
Thanks for your amazing work on this. For the shear amount of new and updated games you added I was actually surprised there weren't more problems !

Regarding 'svc', yes the bootleg 'svcboot' works fine. Is there a reason why it's half the filesize ? That's quite a big difference !

'samsho5' yes it's the warning about illegal carts !

Do you have any idea why the sample files aren't playing ?

Thanks again, and keep up the good work.

**Comment by ZappaUtopia** [comment]
@Wilstorm 
>Wow, a quick question. There's several thousand games and you have tested most of them? That's a huge time saver.

No, I was just talking about the new & updated ones that Arcadez added/fixed !

**Comment by Wilstorm** [comment]
> No, I was just talking about the new & updated ones that Arcadez added/fixed !

Ah ok, I was thinking your fingers must be hurting right about now...since the 4th update that's less than a minute per game if you played around the clock to test them all. I might have tried it....about 30 years ago! ;)

**Comment by Wilstorm** [comment]
@arcadez - Just a quick comment on your whatsnew.txt. It's nicely done and slick to browse through all the updates. Thanks!

**Comment by arcadez** [comment]
Thanks fellas!! hopefully we can get the sample problems ironed out then this core will be
more or less good to go.

**Comment by ZappaUtopia** [comment]
I just built again from source and checked if any roms changed. 'samsho5' and 'samsh5sp' now work correctly.

'mslug5' still doesn't work. Just thought i'd let you know.

**Comment by arcadez** [comment]
Good to hear the Samurai Showdown 5 games now work, what happens exactly with Metal Slug 5
eg does anything happen on boot or does it just fail..??

**Comment by ZappaUtopia** [comment]
'mslug5' doesn't show anything, it just exits back to emulationstation. Not at home at the moment but when I do (in a few hours) i'll build from source and check again (unless someone else confirms the outcome before me).

**Comment by markwkidd** [comment]
Can you post a log?

**Comment by ZappaUtopia** [comment]
I just updated from source, again ! but unfortunately I don't know where to get the new rom (for mslug5). ClrMame reports : 

wrong size: 268-m1_decrypted.bin [wrong: 65536] [right: 262144] [CRC32: 3c0655a7] [not fixed]

The Retropie log currently says : 

Error in RomModule definition: 268-m1.bin out of memory region space
Unable to initialize machine emulation

**Comment by arcadez** [comment]
Alright i'll try one last attempt to fix Metal Slug 5 for now

**Comment by ZappaUtopia** [comment]
The latest commit now says 2 roms are wrong and doesn't work (I don't know if these are real rom sizes) ?

wrong size: 268-m1.bin [wrong: 524288] [right: 262144] [CRC32: 4a5a6e0e] [not fixed]
wrong size: 268-m1_decrypted.bin [wrong: 65536] [right: 262144] [CRC32: 3c0655a7] [not fixed]

I say delete this game as it's too much hassle ! It's no great loss.

**Comment by arcadez** [comment]
Yip i tend to agree ;)

**Comment by arcadez** [comment]
Well Metal Slug 5 and SVC will no longer cause any issues as they are now gone, case closed :)

**Comment by ZappaUtopia** [comment]
for 'titlef' I tried changing the monitor 'A only', 'B only' and 'A and B' but the image is always in the top left part of the screen. This is just for reference, i'm not that bothered. :)

**Comment by arcadez** [comment]
Ah i do remember this being mentioned before with regards to the Sega Multi32 games 
that the dip switch options to switch the monitors around actually dont work for any of
the games in this core.

I'll leave Title Fight supported as someone might do a fix for this someday.

**Comment by Wilstorm** [comment]
One thing that I noticed mainly is when you set "Dip Switches" from the MAME TAB menu it creates a <rom name>,cfg file in the /cfg folder. If you update settings from the F2 Service Menu it creates a <rom name>.nvram file in the /nvram folder.

**Comment by ZappaUtopia** [comment]
Dipswitches work fine. I can change monitors in 'titlef' (Title Fight) to display the left, right or both at the same time but any combination I choose is displayed in the top left half of my TV. (On Raspberry Pi/RetroPie).

**Comment by arcadez** [comment]
@markwkidd i mind someone having this problem a while back with both Title Fight
and Outrunners eg you switch the fake monitor dipswich settings but nothing happens
im sure they also said all the Sega Multi32 games are like this.

The games tend to default to two screen in top of picture if you access the dips and switch
thoes to monitor A or B then the picture should just display one screen and it will fill up
your whole TV screen.

The fake dips since removed from MAME trigger a flag in the system32 video that acts as
the dual or single screen trigger, it's either the dips are not working or the video flag
is not being triggered correctly in the source hence the screens not changing.

**Comment by arcadez** [comment]
@ZappaUtopia just wanna ask did you try B-Rap Boys and Shogun Warriors..?? i remember
someone stating they didn't work correctly a while back but i noticed you dont have them
on your list.

I just wanna know either way

**Comment by grant2258** [comment]
info file might need editing ?

**Comment by grant2258** [comment]
i personally think its the systemname = "Arcade (various)"

might not be

**Comment by arcadez** [comment]
@ZappaUtopia firstly thanks for taking the time to give these a whirl lets start with......

B-Rap Boys and Shogun Warriors
i dunno what to tell ya both of these use a double OKIM6295 the code is intentical in
this core to what i have on the xbox and the sound in these games is perfectly fine.

Backfire!
Again im dumbfounded as i fully tested this one before porting the code across and
there were no graphical niggles that i could see at anyrate, i did play a couple of levels
and everything seemed hunky dory.

Do me a wee favor start a game and after you select a course let me know if it says
8-Way Joystick next to the car, this game does have a wheel option for inputs but
we dont want to see it defaulting to that as it's disabled in the driver.

Basically the driver i ported doesn't support the wheel inputs that might explain why
you cant control the game although it defaulted to joystick on my test core, you can 
always enter service mode right enough and set the game to joystick mode there.

**Comment by ZappaUtopia** [comment]
It does say 8-way joystick on the screen. I can see the wheels moving when I press left/right but accelerate barely moves the car. I pressed F2 and checked but it's already set to joystick.

Regarding the corrupted graphics, it's says 'Push the joystick in the direction of the arrow!' buts it's partly off the top of the screen, the is a vertical line of multicoloured blocks (looks like text or numbers) and a section of the crowd is frozen in place when the screen moves, slowly.

I'm not that fussed to be honest !

**Comment by arcadez** [comment]
Decided to do a little testing myself as per some games that were mentioned as being broken....

Arkanoid (US) (arkanoiu) - Player 2 flipped upside down (vertically)
Dragon Spirit (dspirit) - Player 2 filled upside down (vertically)
Kick (kick) - Player 1 and 2 only flip horizontally (lays on left and right side)

I get the same thing as well i guess it's normal behaviour with the cocktail games

Pig Newton (pignewt) - Crashes black screen

This one booted for me although it has no sound you'd need to use MAME105 to get
the full sound for this one.

D.D. Crew (ddcrew) - Crashes black screen

Like many of the protected Sega 16 / 18 games you'd need to be using MAME95 to be
able to play the likes of DD Crew, FBA will play it though.

**Comment by arcadez** [comment]
Aye im the same not fussed at all about Backfire! it's poor World Rally ripoff, i only added
it to make the Data East changes i did a while back go that little bit further, still having
said that considering it took me an hour or so to do the driver i would have liked for
it to work in this core :)

**Comment by markwkidd** [comment]
I think it's a matter of knowing what games to flag as not working in the
driver versus what games to completely remove from the driver because there
is no known hope of backporting

**Comment by grant2258** [comment]
Well libretro is a problem with Kick (kick) - Player 1 and 2 only flip horizontally (lays on left and right side) this is mirrored wrong changing the flag will fix the problem but other games using this will be the same way. This is something that needs fixed in mame2003 iplementaion

**Comment by arcadez** [comment]
Yeah it's a pity they actually removed the TESTDRIVER handler from driver.c a couple of builds
before MAME78 as if a game was flagged this way it didn't showup in the romlist, which would
be handy for dev's as we would know a game is broken in this core that should work vs a
game which didn't work anyway at this point in the MAME dev cycle.

**Comment by grant2258** [comment]
the game does work if you change it to baddump like hydra thats how mame latest does it pignewton that is

**Comment by ZappaUtopia** [comment]
`sandscrp` Sand Scorpion no longer has in-game sound in lr-mame2003-plus, it worked fine in lr-mame2003.

**Comment by grant2258** [comment]
for the mirrored screens that arent cocktail flips post here #81

**Comment by arcadez** [comment]
Ah so it worked before then i must mixed up Sand Scorpian with another game, ok i'll
look at the sound so what got broke

**Comment by arcadez** [comment]
Alright i just tried Sand Scorpian and yeah there is a problem with the sound when the levels
start there is none :), although sometimes it pops back on as you progress through the level
or when you do an ingame rest.

i just compaired the mame2003 driver vs the one that was updated for this core and whatya
know everything barring some small video handlers is the exactly the same, this might be
one of those little mysteries i might not be able to solve as i was hoping to find something
off with the sound code.

At the end of the day should a fix not be found we might have to think Sand Scorpian 
with buggy sound now vs all those newly supported and way better games is a fair tradeoff..??

**Comment by grant2258** [comment]
well lets try see what went wrong must be fixable just leave it till we look

**Comment by grant2258** [comment]
what the romname of sand scorpian ?

**Comment by markwkidd** [comment]
sandscrp

**Comment by markwkidd** [comment]
@ZappaUtopia do you feel ok about closing this now?

**Comment by grant2258** [comment]
@ZappaUtopia @markwkidd  @arcadez there is some miss infromation about sand scorpian. I tired on mame078 mame2003 and mame2003 plus none have sound working

**Comment by ZappaUtopia** [comment]
On RetroPie using lr-mame2003 I have full sound in-game in `sandscrp` (Sand Scorpion). On lr-mame2003-plus there is no sound in-game. I'm happy to close this after your thoughts on this post !

**Comment by arcadez** [comment]
Im thinking a problem with the IRQ commands.? but yeah cant rule out the banks, anyway i've flagged
it now pending further investigation but it's not a great game so not high on my priority list.

**Comment by ZappaUtopia** [comment]
@grant2258 There is sound (in-game) in lr-mame2003 (Retropie) but almost silent (in-game) in lr-mame2003plus. I'll close this anyway !
